src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+        const cache = this.getCache();
+        const cacheKey = this.getCacheKey(id);
+        return cache.get(cacheKey);
+    }
+
+    /**
+     * Sets the items in the cache
+     *
+     * @param {string} id - the cache id
+     * @param {Array} items - the feed items to cache
+     */
+    setCachedItems(id: string, items: FeedItems) {
+        const cache = this.getCache();
+        const cacheKey = this.getCacheKey(id);
+        cache.set(cacheKey, {
+            hasError: !!this.hasError,
+            items
+        });
+    }
+
+    /**
+     * Gets the feed items
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     */
+    feedItems(file: BoxItem, successCallback: Function, errorCallback: Function): void {
+        const { id } = file;
+
+        if (!id) {
+            throw getBadItemError();
+        }
+
+        const cachedItems = this.getCachedItems(id);
+        if (cachedItems) {
+            const { hasError, items } = cachedItems;
+            if (hasError) {
+                errorCallback(items);
+            } else {
+                successCallback(items);
+            }
+
+            return;
+        }
+
+        this.id = id;
+        this.hasError = false;
+        const versionsPromise = this.fetchVersions();
+        const commentsPromise = this.fetchComments();
+        const tasksPromise = this.fetchTasks();
+
+        Promise.all([versionsPromise, commentsPromise, tasksPromise]).then((feedItems) => {
+            const versions: ?FileVersions = feedItems[0];
+            const versionsWithRestoredVersion = this.addCurrentVersion(versions, file);
+            const unsortedFeedItems = [versionsWithRestoredVersion, ...feedItems.slice(1)];
+            const sortedFeedItems = this.sortFeedItems.apply(null, unsortedFeedItems);
+            this.setCachedItems(id, sortedFeedItems);
+            if (!this.isDestroyed()) {
+                if (this.hasError) {
+                    errorCallback(sortedFeedItems);
+                } else {
+                    successCallback(sortedFeedItems);
+                }
+            }
+        });
+    }
+
+    /**
+     * Sort valid feed items, descending by created_at time.
+     *
+     * @param {Array<?Comments | ?Tasks | ?FileVersions>} args - Arguments list of each item container
+     * type that is allowed in the feed.
+     * @return {Array<?Comments | ?Tasks | ?FileVersions>} the sorted feed items
+     */
+    sortFeedItems(...args: Array<?Comments | ?Tasks | ?FileVersions>): FeedItems {
+        const feedItems = [];
+        args.forEach((itemContainer) => {
+            if (itemContainer) {
+                feedItems.push(...itemContainer.entries);
+            }
+        });
+        return feedItems.sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at));
```

Comment:
Agreed, but I wanted to avoid refactoring each and every method. Some of these methods are direct copy + paste from existing files. I can take a look at this function anyways